content/shmem_team_broadcast.tex

```diff
@@ -0,0 +1,90 @@
+\apisummary{
+    Broadcasts a block of data from one \ac{PE} to one or more destination
+    \acp{PE}.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_team\_broadcast32}@(shmem_team_t team, void *dest, const void *source, size_t nelems, int PE_root);
+void @\FuncDecl{shmem\_team\_broadcast64}@(shmem_team_t team, void *dest, const void *source, size_t nelems, int PE_root);
+\end{Csynopsis}
+
+\begin{apiarguments}
+
+\apiargument{IN}{team}{A valid \openshmem team handle to a team which has been created without
+    disabling support for collective operations.}
+\apiargument{OUT}{dest}{A symmetric data object. See the table below in this description
+    for allowable types} 
+\apiargument{IN}{source}{A symmetric data object that can be of any data type
+    that is permissible for the \dest{} argument.}
+\apiargument{IN}{nelems}{The number of elements in \source{}.  For
+    \FUNC{shmem\_team\_broadcast32}, this is the number of
+    32-bit halfwords. nelems must be of type \CTYPE{size\_t} in \Cstd.}
+\apiargument{IN}{PE\_root}{Zero-based ordinal of the \ac{PE}, with respect to
+    the team, from which the data is copied. Must be greater than or equal to
+    0 and less than N-1, where N is the result of calling \FUNC{shmem\_team\_n\_pes(team)}.
+    \VAR{PE\_root} must be of type \CTYPE{int}.} 
+
+\end{apiarguments}
+
+\apidescription{   
+    \openshmem broadcast routines are collective routines over an existing team.
+    They copy data object \source{} on the processor specified by \VAR{PE\_root}
+    and store the values at \dest{} on the other \acp{PE} that are members of the
+    team. The data is not copied to the \dest{} area on the root \ac{PE}.
+    
+    If the team has been created with the \LibConstRef{SHMEM\_TEAM\_NOCOLLECTIVE} option,
+    it will not have the required support structures to complete this routine. If
+    such a team is passed to this or any other team collective routine, the behavior
+    is undefined.
+
+    As with all \openshmem routines where the operation occurs for a given team -
+    either when the team is an argument to the routine, or when the team is an attribute
+    of the context argument to a routine - the \ac{PE} numbers are relative to the team,
+    and must be in the range of 0 to N-1, where N is the result of \FUNC{shmem\_team\_n\_pes(team)}.
+    
+    The values of the argument \VAR{PE\_root} must be the same value on all \acp{PE} in
+    the team. The same \dest{} and \source{} data objects must be passed by all \acp{PE}
```

Comment:
The requirement that `dest` and `source` be same is currently missing in spec 1.4 - is that a known issue?
Also, why is `dest` required to be the same? Also missing from spec 1.4, can `nelems` be different across calling PEs?

Reply:
Maybe I misunderstand the question? I think this requirement is in 1.4 for broadcast: 

"The values of arguments PE_root, PE_start, logPE_stride, and PE_size must be the same value on all PEs in the active set. The same dest and source data objects and the same pSync work array must be passed by all PEs in the active set"

I think the requirement is there because source and dest have to be symmetric. 

As for nelems, there is no stated requirement in 1.4, but it seems it should have to be the same, since the source and dest arrays have to be the same size and symmetric.

Reply:
My bad, I missed this - `The same dest and source data objects and the same pSync work array must be passed by all PEs in the active set`